src/View/Helper/FormHelper.php

```diff
@@ -410,183 +414,286 @@ public function control(string $fieldName, array $options = []): string
             case 'datetime':
             case 'date':
             case 'time':
-                $options['label']['templateVars']['groupId'] =
-                $options['templateVars']['groupId'] =
-                    $this->_domId($fieldName . '-group-label');
-
-                $options['templates']['label'] = $this->templater()->get('datetimeLabel');
-                $options['templates']['inputContainer'] = $this->templater()->get('datetimeContainer');
-                $options['templates']['inputContainerError'] = $this->templater()->get('datetimeContainerError');
+                $options = $this->_dateTimeOptions($fieldName, $options);
                 break;
 
             case 'checkbox':
-                if (!$custom) {
-                    $options['label'] = $this->injectClasses('form-check-label', (array)$options['label']);
-                    $options = $this->injectClasses('form-check-input', $options);
-                } else {
-                    $options['label'] = $this->injectClasses('custom-control-label', (array)$options['label']);
-                    $options = $this->injectClasses('custom-control-input', $options);
-
-                    if ($this->_align === 'horizontal') {
-                        $options['templates']['checkboxFormGroup'] = $this->templater()->get('customCheckboxFormGroup');
-                    } else {
-                        $options['templates']['checkboxContainer'] = $this->templater()->get('customCheckboxContainer');
-                        $containerError = $this->templater()->get('customCheckboxContainerError');
-                        $options['templates']['checkboxContainerError'] = $containerError;
-                    }
-                }
+            case 'radio':
+            case 'select':
+            case 'file':
+                $function = '_' . $options['type'] . 'Options';
+                $options = $this->{$function}($fieldName, $options);
+                break;
 
-                if ($this->_align === 'horizontal') {
-                    $inline = false;
+            case 'range':
+                if ($options['custom']) {
+                    $options['injectFormControl'] = false;
+                    $options = $this->injectClasses('custom-range', $options);
                 }
+                break;
```

Comment:
Is that an oversight, or did you leave it because it's so short?

Reply:
The latter :)